llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1272,9 +1272,12 @@ Value *SPIRVToLLVM::transShiftLogicalBitwiseInst(SPIRVValue *bv, BasicBlock *bb,
   if (shift->getType()->isIntOrIntVectorTy())
     shift = getBuilder()->CreateZExtOrTrunc(shift, base->getType());
 
-  auto inst = BinaryOperator::Create(bo, base, shift, bv->getName(), bb);
+  Instruction *inst = BinaryOperator::Create(bo, base, shift, bv->getName(), bb);
   setFastMathFlags(inst);
 
+  if (isShiftOpCode(op))
+    inst = new FreezeInst(inst, "shift.freeze", bb);
```

Comment:
Some of the code above already uses the builder, can we just align with that here as well?